File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        if (cacheSize == 0) {
            return cities.length * 5;
        }
        Queue<String> q = new LinkedList<>();
        int answer = 0;
        
        for(String city : cities) {
            city = city.toLowerCase();
            
            if (q.contains(city)) {
                q.remove(city);
                q.add(city);
                answer++;
            } else {
                answer += 5;
                if (q.size() == cacheSize) {
                    q.poll();       
                }
                q.add(city);
            }
        }
        
        return answer;
        
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque
def solution(cacheSize, cities):
    cache = deque(maxlen=cacheSize)
    answer = 0

    for city in map(lambda x: x.lower(), cities):
        if city in cache:
            cache.remove(city)
            cache.append(city)
            answer += 1
            continue
            
        cache.append(city)
        answer += 5
    return answer